pkg/store/store.go

```diff
@@ -50,18 +50,50 @@ func NewOpenTSDBStore(logger log.Logger, client opentsdb.ClientContext, reg prom
 	return store
 }
 
+type internalMetrics struct {
+	numberOfOpenTSDBMetrics *prometheus.GaugeVec
+	lastUpdateOfOpenTSDBMetrics prometheus.Gauge
+	openTSDBLatency         *prometheus.HistogramVec
+	servedDatapoints        prometheus.Counter
+}
+
+func newInternalMetrics(reg prometheus.Registerer) internalMetrics {
+	m := internalMetrics{
+		openTSDBLatency: prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{Name: "geras_opentsdb_request_latency_seconds",
+			Buckets: []float64{
+				0.01, 0.05, 0.1, 0.5, 1, 5, 10, 20, 50,
+			}},
+			[]string{"endpoint", "status"}),
+		servedDatapoints: prometheus.NewCounter(prometheus.CounterOpts{Name: "geras_served_datapoints_total"}),
+		numberOfOpenTSDBMetrics: prometheus.NewGaugeVec(prometheus.GaugeOpts{Name: "geras_cached_metrics"},
+			[]string{"type"}),
+		lastUpdateOfOpenTSDBMetrics: prometheus.NewGauge(prometheus.GaugeOpts{Name: "geras_metrics_cache_update_time"}),
+	}
+	if reg != nil {
+		reg.MustRegister(m.openTSDBLatency)
+		reg.MustRegister(m.servedDatapoints)
+		reg.MustRegister(m.numberOfOpenTSDBMetrics)
+		reg.MustRegister(m.lastUpdateOfOpenTSDBMetrics)
+	}
+	return m
+}
+
 func (store *OpenTSDBStore) updateMetrics(ctx context.Context, logger log.Logger) {
 	events := trace.NewEventLog("store.updateMetrics", "")
 
 	fetch := func() {
 		events.Printf("Refresh metrics")
 		tr := trace.New("store.updateMetrics", "fetch")
 		defer tr.Finish()
+		ctx, cancel := context.WithTimeout(ctx, 60 * time.Second)
```

Comment:
60sec seems reasonable enough to hard-code I guess

Reply:
Yeah, can always revisit, at least we have metrics and tracing now to actually see ;-)